components/shared/base/generic_channel.rs

```diff
@@ -110,6 +113,31 @@ where
     }
 }
 
+impl<T> Serialize for GenericReceiver<T>
+where
+    T: for<'de> Deserialize<'de> + Serialize,
+{
+    fn serialize<S: Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        match self {
+            GenericReceiver::Ipc(i) => i.serialize(s),
+            GenericReceiver::Crossbeam(_) => unreachable!(),
```

Comment:
Please write a message here:

```suggestion
            GenericReceiver::Crossbeam(_) => unreachable!(),
```

Reply:
done, please check. Also used the same message for the serialization of the GenericSender.